### PR TITLE
Release version 4.4.1

### PR DESCRIPTION
**Pull request by rcurtin**
This automatically-generated pull request adds the commits necessary to make the 4.4.1 release.

Once the PR is merged, mlpack-bot will tag the release as HEAD~1 (so that it doesn't include the new HISTORY block) and publish it.

When you merge this PR, be sure to merge it using a *rebase*.

### Changelog

 * Distribute STB headers as part of R package (#3724, #3726).

 * Added OpenMP parallelization to Hamerly, Naive, and Elkan k-means (#3761, #3762, #3764).
 
 * Added OpenMP support for fast approximation (#3685).

 * Implemented the Find and Fill algorithm into the Dropout Layer and added OpenMP support (#3684).

 * Update Python bindings to support NumPy 2.x (#3752).

 * Bump minimum Armadillo version to 10.8 (#3760).

 * Adapt `NearestInterpolation` ANN layer to new Layer Inteface (#3768).

 * Add support for arbitrary matrix types to `Radical` and deprecate `Radical::DoRadical()` in favor of `Radical::Apply()` (#3787).

### CODE REVIEW

**Review by shrit — APPROVED** [review]

**Comment by conradsnicta** [comment]
@rcurtin Suggest to call this release 4.5.0 rather than 4.4.1.

Bumping the patch level (the last .1) usually indicates bug fixes, with no additions or other improvements.  This is definitely not the case here.

Instead, it's a decent minor release, and calling it 4.4.1 undersells the improvements and changes.  It deserves to be called 4.5.0.

People like shiny things and bigger numbers, so from a marketing perspective it'd be useful to send the right message.

File: HISTORY.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 _????-??-??_
 
+
+## mlpack 4.4.1
```
Comment (conradsnicta) [inline HISTORY.md:8]: suggest to bump the version to 4.5.0, as per my earlier comment

**Review by conradsnicta — CHANGES_REQUESTED** [review]
suggest to bump the version to 4.5.0, as per my earlier comment

**Comment by rcurtin** [comment]
Fair enough, opened #3798 to bump to 4.5.0 instead.

**Pull request closed without merging**